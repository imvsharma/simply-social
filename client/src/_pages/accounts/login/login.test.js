import React, { Component } from 'react';
import { shallow } from 'enzyme';
import Login from './login';

describe('Login Component', () => {
    it('should render login component', () => {
        const wrapper = shallow(<Login />)
    });

    it('should render initial layout', () => {
        const component = shallow(<Login />);
        expect(component.getElements()).toMatchSnapshot();
    })

    it('should have email state', () => {
        const component = shallow(<Login />);
        const form = component.find('input[type="email"]');
        /* form.props().onChange({
            target: {
                name: 'myName',
                value: 'myValue'
            }
        }) */
        expect(component.state('email')).toBeDefined()
    })

    it('should have update email onchange event', () => {
        const component = shallow(<Login />);
        const form = component.find('input[type="email"]');

        const mockEvent = {
            target: {
              name: "email",
              value: "abc@xyz.com"
            }
        };

        component.instance().handleChange(mockEvent);
        expect(component.state('email')).toEqual("abc@xyz.com");
    })
})
