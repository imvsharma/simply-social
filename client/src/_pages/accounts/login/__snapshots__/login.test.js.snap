// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should render initial layout 1`] = `
Array [
  <Context.Provider
    value={null}
  >
    <Login
      dispatch={[Function]}
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    />
  </Context.Provider>,
]
`;
